test/integration/apiserver.go

```diff
@@ -164,9 +166,179 @@ func newAPIServer(f *testutil.Framework) func(t *testing.T) {
+				Name:      "test-offering-2",
+				Namespace: testName,
+				Labels: map[string]string{
+					"test-label": "offering2",
+				},
+			},
+			Spec: catalogv1alpha1.OfferingSpec{
+				Metadata: catalogv1alpha1.OfferingMetadata{
+					Description: "Test Offering",
+					DisplayName: "Test Offering",
+				},
+				Provider: catalogv1alpha1.ObjectReference{
+					Name: "test-provider",
+				},
+				CRD: catalogv1alpha1.CRDInformation{
+					Name:     "test-crd",
+					APIGroup: "test-crd-group",
+					Kind:     "test-kind",
+					Plural:   "test-plural",
+					Versions: []catalogv1alpha1.CRDVersion{
+						{
+							Name: "test-version",
+							Schema: &apiextensionsv1.CustomResourceValidation{
+								OpenAPIV3Schema: &apiextensionsv1.JSONSchemaProps{
+									Properties: map[string]apiextensionsv1.JSONSchemaProps{
+										"apiVersion": {Type: "string"},
+									},
+									Type: "object",
+								},
+							},
+						},
+					},
+					Region: catalogv1alpha1.ObjectReference{
+						Name: "test-region",
+					},
+				},
+			},
+		}
+		require.NoError(t, managementClient.Create(ctx, offering1))
+		require.NoError(t, managementClient.Create(ctx, offering2))
+
+		client := apiserverv1.NewOfferingServiceClient(conn)
+		offeringCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		t.Cleanup(cancel)
+		// list offerings with limit and continuation token.
+		require.NoError(t, wait.PollUntil(time.Second, func() (done bool, err error) {
+			offerings, err := client.List(offeringCtx, &apiserverv1.OfferingListRequest{
+				Tenant: testName,
+				Limit:  1,
+			})
+			if err != nil {
+				return false, err
+			}
+			assert.Len(t, offerings.Items, 1)
+			testutil.LogObject(t, offerings)
+			offerings, err = client.List(offeringCtx, &apiserverv1.OfferingListRequest{
+				Tenant:   testName,
+				Limit:    1,
+				Continue: offerings.Continue,
+			})
+			if err != nil {
+				return false, err
+			}
+			assert.Len(t, offerings.Items, 1)
+			testutil.LogObject(t, offerings)
+			return true, nil
+		}, offeringCtx.Done()))
+
+		// get offering
+		require.NoError(t, wait.PollUntil(time.Second, func() (done bool, err error) {
+			offering, err := client.Get(offeringCtx, &apiserverv1.OfferingRequest{
+				Tenant: testName,
+				Name:   "test-offering-1",
+			})
+			if err != nil {
+				return false, err
+			}
+			expectedResult := &apiserverv1.Offering{
+				Name: "test-offering-1",
+				Metadata: &apiserverv1.OfferingMetadata{
+					Description: "Test Offering",
+					DisplayName: "Test Offering",
+				},
+				Provider: &apiserverv1.ObjectReference{
+					Name: "test-provider",
+				},
+				Crd: &apiserverv1.CRDInformation{
+					Name:     "test-crd",
+					ApiGroup: "test-crd-group",
+					Kind:     "test-kind",
+					Plural:   "test-plural",
+					Versions: []*apiserverv1.CRDVersion{
+						{
+							Name:   "test-version",
+							Schema: "{\"openAPIV3Schema\":{\"type\":\"object\",\"properties\":{\"apiVersion\":{\"type\":\"string\"}}}}",
```

Comment:
you can use golang raw string here. https://golang.org/ref/spec#String_literals

Reply:
thanks！